File: Comms Protocol Java/Comms Protocol Java/comms/protocol/java/LinearAcceleration_RawDataPacket.java
package comms.protocol.java;

public class LinearAcceleration_RawDataPacket extends AndroidDataPacket {

	private static int numberOfValues = 3;

	public LinearAcceleration_RawDataPacket() {
		super(ValidPacketTypes.linear_acceleration_raw_data_packet, numberOfValues);
	}

	public LinearAcceleration_RawDataPacket(byte[] payload) throws Exception {
		super(ValidPacketTypes.linear_acceleration_raw_data_packet, numberOfValues, payload);
	}

	public LinearAcceleration_RawDataPacket(DataPacket packet) throws Exception {
		super(ValidPacketTypes.linear_acceleration_raw_data_packet, numberOfValues, packet);
	}
}

File: Comms Protocol CSharp/Comms Protocol CSharp/LinearAcceleration_RawDataPacket.cs

namespace Comms_Protocol_CSharp
{
    public class LinearAcceleration_RawDataPacket : AndroidDataPacket
    {
        private static int numberOfVals = 3;

        public LinearAcceleration_RawDataPacket() : base(ValidPacketTypes.linear_acceleration_raw_data_packet, numberOfVals)
        {

        }

        public LinearAcceleration_RawDataPacket(byte[] payload) : base(ValidPacketTypes.linear_acceleration_raw_data_packet, numberOfVals, payload)
        {

        }

        public LinearAcceleration_RawDataPacket(DataPacket packet) : base(ValidPacketTypes.linear_acceleration_raw_data_packet, numberOfVals, packet)
        {

        }

        public XYZSensor GetAcceleration()
        {
            AndroidSensor sensor = DeSerialize();
            return new XYZSensor(sensor.GetValues());
        }
    }
}
